### ISSUE TITLE
Couldn't execute "/gnu/store/8llinghxcslyxmvszjlvlxbv411vra0x-next-gtk-webkit-1.1.0/bin/next-gtk-webkit": No such file or directory

### ISSUE THREAD

**Issue by bmillare** [atlas-engineer/next#160]
sha256sum ~/Downloads/next-linux-gtk-webkit.tar.lz 
e7789c4c110746f6922683071b3ee5ecbd596cc3d2bb542197592b9e3b809400  /home/bmillare/Downloads/next-linux-gtk-webkit.tar.lz
# this matches the website instructions: https://next.atlas.engineer/download
tar --lzip -xvpf ~/Downloads/next-linux-gtk-webkit.tar.lz 
./bin/next
WARNING: Setting locale failed.
  Check the following variables for correct values:
  LANG=en_US.UTF-8
Unhandled SIMPLE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                    {10005885B3}>:
  Couldn't execute "/gnu/store/8llinghxcslyxmvszjlvlxbv411vra0x-next-gtk-webkit-1.1.0/bin/next-gtk-webkit": No such file or directory

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10005885B3}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<SIMPLE-ERROR "Couldn't execute ~S: ~A" {10037921E3}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* #<SIMPLE-ERROR "Couldn't execute ~S: ~A" {10037921E3}>)
2: (INVOKE-DEBUGGER #<SIMPLE-ERROR "Couldn't execute ~S: ~A" {10037921E3}>)
3: (UIOP/IMAGE:HANDLE-FATAL-CONDITION #<SIMPLE-ERROR "Couldn't execute ~S: ~A" {10037921E3}>)
4: (SB-KERNEL::%SIGNAL #<SIMPLE-ERROR "Couldn't execute ~S: ~A" {10037921E3}>)
5: (ERROR "Couldn't execute ~S: ~A" "/gnu/store/8llinghxcslyxmvszjlvlxbv411vra0x-next-gtk-webkit-1.1.0/bin/next-gtk-webkit" "No such file or directory")
6: (RUN-PROGRAM "/gnu/store/8llinghxcslyxmvszjlvlxbv411vra0x-next-gtk-webkit-1.1.0/bin/next-gtk-webkit" ("--port" "8082" "--core-socket" "http://localhost:8081/RPC2") :ENV NIL :ENVIRONMENT NIL :WAIT NIL :SEARCH T :PTY NIL :INPUT NIL :IF-INPUT-DOES-NOT-EXIST :ERROR :OUTPUT #P"/tmp/next-gtk-webkit.log" :IF-OUTPUT-EXISTS :APPEND :ERROR :OUTPUT :IF-ERROR-EXISTS :APPEND :STATUS-HOOK NIL :EXTERNAL-FORMAT :UTF-8 :DIRECTORY NIL)
7: (UIOP/LAUNCH-PROGRAM:LAUNCH-PROGRAM ("/gnu/store/8llinghxcslyxmvszjlvlxbv411vra0x-next-gtk-webkit-1.1.0/bin/next-gtk-webkit" "--port" "8082" "--core-socket" "http://localhost:8081/RPC2") :OUTPUT #P"/tmp/next-gtk-webkit.log" :ERROR-OUTPUT :OUTPUT)
8: ((:METHOD NEXT::RUN-PROGRAM (NEXT::PORT)) #<NEXT::PORT {1001CB0BF3}>) [fast-method]
9: (NEXT:START :WITH-PLATFORM-PORT-P T)
10: (NEXT-EXEC:MAIN)
11: ((LAMBDA NIL :IN UIOP/IMAGE:RESTORE-IMAGE))
12: (UIOP/IMAGE:CALL-WITH-FATAL-CONDITION-HANDLER #<CLOSURE (LAMBDA NIL :IN UIOP/IMAGE:RESTORE-IMAGE) {10035CE81B}>)
13: ((FLET SB-UNIX::BODY :IN SAVE-LISP-AND-DIE))
14: ((FLET "WITHOUT-INTERRUPTS-BODY-27" :IN SAVE-LISP-AND-DIE))
15: ((LABELS SB-IMPL::RESTART-LISP :IN SAVE-LISP-AND-DIE))

unhandled condition in --disable-debugger mode, quitting

**Comment by Ambrevar** [comment]
Have you followed the instructions from the download page?

- You can either extract the archive to root (`/`), it will only create two
  folders, `/bin` and `/gnu/store`.
  
- Or you can enable name spacing, with one of the following commands:
  
	# sysctl -w kernel.unprivileged_userns_clone=1
  # echo 0 > /proc/sys/kernel/userns_restrict
  
  See your distribution for more details on this.

If none of the above works for you, you could try some of the suggestions from
the developer's README:

https://github.com/atlas-engineer/next/tree/master/documents#run

**Comment by Carpetsmoker** [comment]
I had the same issue on my Arch Linux machine; the problem seems to be that this directory doesn't exist:

```
[~/next]% lc ./gnu/store/ | grep web
dr-xr-xr-x  6 martin martin 4.0K Jan  1  1970 45xxf6ga626v6wgf30h2zk787qx4grif-libwebp-1.0.1/
dr-xr-xr-x  7 martin martin 4.0K Jan  1  1970 86lf85302y2lifwch723w82w5yihhdh8-webkitgtk-2.20.5/
dr-xr-xr-x  4 martin martin 4.0K Jan  1  1970 ildn4l2wr0y7irm2536dnp88hhrdphsz-next-gtk-webkit-1.1.0/
```

I fixed it with:

```
[~/next]% cd gnu/store
[~/next/gnu/store]% ln -s ildn4l2wr0y7irm2536dnp88hhrdphsz-next-gtk-webkit-1.1.0 8llinghxcslyxmvszjlvlxbv411vra0x-next-gtk-webkit-1.1.0
```

And now it runs.

I also have the "setting locale failed" warnings; no idea if that's related (or even why I'm getting them :thinking:)

```
[~/next]% ./bin/next                                                                                                                  i:x270:20:16
WARNING: Setting locale failed.
  Check the following variables for correct values:
  LC_CTYPE=en_US.UTF-8
  LANG=en_US.UTF-8
  LC_COLLATE=en_US.UTF-8
Unhandled SIMPLE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                    {10005885B3}>:
  Couldn't execute "/gnu/store/8llinghxcslyxmvszjlvlxbv411vra0x-next-gtk-webkit-1.1.0/bin/next-gtk-webkit": No such file or directory

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10005885B3}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<SIMPLE-ERROR "Couldn't execute ~S: ~A" {10037921E3}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* #<SIMPLE-ERROR "Couldn't execute ~S: ~A" {10037921E3}>)
2: (INVOKE-DEBUGGER #<SIMPLE-ERROR "Couldn't execute ~S: ~A" {10037921E3}>)
3: (UIOP/IMAGE:HANDLE-FATAL-CONDITION #<SIMPLE-ERROR "Couldn't execute ~S: ~A" {10037921E3}>)
4: (SB-KERNEL::%SIGNAL #<SIMPLE-ERROR "Couldn't execute ~S: ~A" {10037921E3}>)
5: (ERROR "Couldn't execute ~S: ~A" "/gnu/store/8llinghxcslyxmvszjlvlxbv411vra0x-next-gtk-webkit-1.1.0/bin/next-gtk-webkit" "No such file or directory")
6: (RUN-PROGRAM "/gnu/store/8llinghxcslyxmvszjlvlxbv411vra0x-next-gtk-webkit-1.1.0/bin/next-gtk-webkit" ("--port" "8082" "--core-socket" "http://localhost:8081/RPC2") :ENV NIL :ENVIRONMENT NIL :WAIT NIL :SEARCH T :PTY NIL :INPUT NIL :IF-INPUT-DOES-NOT-EXIST :ERROR :OUTPUT #P"/tmp/next-gtk-webkit.log" :IF-OUTPUT-EXISTS :APPEND :ERROR :OUTPUT :IF-ERROR-EXISTS :APPEND :STATUS-HOOK NIL :EXTERNAL-FORMAT :UTF-8 :DIRECTORY NIL)
7: (UIOP/LAUNCH-PROGRAM:LAUNCH-PROGRAM ("/gnu/store/8llinghxcslyxmvszjlvlxbv411vra0x-next-gtk-webkit-1.1.0/bin/next-gtk-webkit" "--port" "8082" "--core-socket" "http://localhost:8081/RPC2") :OUTPUT #P"/tmp/next-gtk-webkit.log" :ERROR-OUTPUT :OUTPUT)
8: ((:METHOD NEXT::RUN-PROGRAM (NEXT::PORT)) #<NEXT::PORT {1001CB0BF3}>) [fast-method]
9: (NEXT:START :WITH-PLATFORM-PORT-P T)
10: (NEXT-EXEC:MAIN)
11: ((LAMBDA NIL :IN UIOP/IMAGE:RESTORE-IMAGE))
12: (UIOP/IMAGE:CALL-WITH-FATAL-CONDITION-HANDLER #<CLOSURE (LAMBDA NIL :IN UIOP/IMAGE:RESTORE-IMAGE) {10035CE80B}>)
13: ((FLET SB-UNIX::BODY :IN SAVE-LISP-AND-DIE))
14: ((FLET "WITHOUT-INTERRUPTS-BODY-27" :IN SAVE-LISP-AND-DIE))
15: ((LABELS SB-IMPL::RESTART-LISP :IN SAVE-LISP-AND-DIE))

unhandled condition in --disable-debugger mode, quitting
```

**Comment by Ambrevar** [comment]
Thanks for the feedback.  I think I know what's happening, I'll fix it later
today.

Regarding the locale issue, don't worry, you can fix it with some tweaks on your
system but it's harmless, plus Next does not support locales yet :)

**Comment by Ambrevar** [comment]
@gumblr: Does Next fully work for you with the symlink trick?

**Comment by Ambrevar** [comment]
1.2 is out!  I've just uploaded a new Guix pack which should now fix this issue.  Can anyone confirm?

**Comment by Ambrevar** [comment]
This should be fixed in Next 1.2.1. Feel free to re-open if you still experience the issue.
